broker-core/src/test/java/io/zeebe/broker/logstreams/delete/LeaderLogStreamDeletionTest.java

```diff
@@ -0,0 +1,145 @@
+  private final ServiceContainerRule serviceContainer = new ServiceContainerRule(actorScheduler);
+
+  @Rule public RuleChain chain = RuleChain.outerRule(actorScheduler).around(serviceContainer);
+
+  @Mock ExporterManagerService mockExporterManagerService;
+  @Mock LogStream mockLogStream;
+  @Mock LogBlockIndex mockLogBlockIndex;
+  @Mock LogStorage mockLogStorage;
+  @Mock LogBlockIndexContext mockBlockIndexContext;
+
+  private LeaderLogStreamDeletionService deletionService;
+
+  private static final long POSITION_TO_DELETE = 6L;
+  private static final long ADDRESS_TO_DELETE = 55L;
+
+  private LogStream prepareMockLogStream() {
+    mockLogStream = mock(LogStream.class);
+    mockLogBlockIndex = mock(LogBlockIndex.class);
+    mockLogStorage = mock(LogStorage.class);
+    mockBlockIndexContext = mock(LogBlockIndexContext.class);
+
+    when(mockLogStream.getLogBlockIndex()).thenReturn(mockLogBlockIndex);
+    when(mockLogStream.getLogStorage()).thenReturn(mockLogStorage);
+    when(mockLogBlockIndex.lookupBlockAddress(mockBlockIndexContext, POSITION_TO_DELETE))
+        .thenReturn(ADDRESS_TO_DELETE);
+    when(mockLogBlockIndex.lookupBlockAddress(mockBlockIndexContext, -1L)).thenReturn(-1L);
+    when(mockLogBlockIndex.createLogBlockIndexContext()).thenReturn(mockBlockIndexContext);
+
+    return mockLogStream;
+  }
+
+  private void createLogStreamDeletionService(LogStream logStream) {
+    deletionService = new LeaderLogStreamDeletionService(logStream);
+    serviceContainer
+        .get()
+        .createService(
+            EngineServiceNames.leaderLogStreamDeletionService(PARTITION_ID), deletionService)
+        .install()
+        .join();
+    final Injector<ExporterManagerService> exporterManagerInjector =
+        deletionService.getExporterManagerInjector();
+
+    mockExporterManagerService = mock(ExporterManagerService.class);
+    exporterManagerInjector.inject(mockExporterManagerService);
+    deletionService.start(mock(ServiceStartContext.class));
+  }
+
+  @Before
+  public void setup() {
+    final LogStream logStream = prepareMockLogStream();
+    createLogStreamDeletionService(logStream);
+  }
+
+  @Test
+  public void shouldDeleteWithDelayedExporter() {
+    // given
+    when(mockExporterManagerService.getLowestExporterPosition()).thenReturn(POSITION_TO_DELETE);
+
+    // when
+    deletionService.delete(POSITION_TO_DELETE + 2);
+
+    // then
+    verify(mockLogBlockIndex, times(1))
+        .lookupBlockAddress(mockBlockIndexContext, POSITION_TO_DELETE);
+    verify(mockLogBlockIndex, times(1))
+        .deleteUpToPosition(mockBlockIndexContext, POSITION_TO_DELETE);
+    verify(mockLogStorage, times(1)).delete(ADDRESS_TO_DELETE);
+  }
+
+  @Test
+  public void shouldDeleteWithNoExporter() {
+    // given
+    when(mockExporterManagerService.getLowestExporterPosition()).thenReturn(Long.MAX_VALUE);
+
+    // when
+    deletionService.delete(POSITION_TO_DELETE);
+
+    // then
+    verify(mockLogBlockIndex, times(1))
```

Comment:
This is more implementation detail how the deletion works internally in the logstream. I think it is enough to verify that the logstream delete is called with this position.